sample_project/Assets/SampleViewer/Samples/Geometry/Geometries.cs

```diff
@@ -194,10 +211,101 @@ private void OnLeftClickEnd(InputAction.CallbackContext context)
+
+            if (Physics.Raycast(ray, out hit))
+            {
+                hit.point += new Vector3(0, MarkerHeight, 0);
+                var hitPoint = arcGISMapComponent.EngineToGeographic(hit.point);
+
+                if (isEnvelopeMode)
+                {
+                    startPoint = hitPoint;
+                    isDragging = true;
+                }
+                else
+                {
+                    if (isPolygonMode)
+                    {
+                        //clear interpolation points of last segments 
+                        foreach (var point in lastToStartInterpolationPoints)
+                        {
+                            Destroy(point);
+                            continue;
+                        }
+                        lastToStartInterpolationPoints.Clear();
+                    }
+
+                    var lineMarker = Instantiate(LineMarker, hit.point, Quaternion.identity, arcGISMapComponent.transform);
+                    var location = lineMarker.GetComponent<ArcGISLocationComponent>();
+                    location.enabled = true;
+                    location.Position = hitPoint;
+                    location.Rotation = new ArcGISRotation(0, 90, 0);
+
+                    if (stops.Count > 0)
+                    {
+                        GameObject lastStop = stops.Peek();
+                        var lastPoint = lastStop.GetComponent<ArcGISLocationComponent>().Position;
+                        if (isPolylineMode)
+                        {
+                            // Calculate distance from last point to this point, and add to the total distance calculation.
+                            calculation += ArcGISGeometryEngine.DistanceGeodetic(lastPoint, hitPoint, currentLinearUnit, new ArcGISAngularUnit(ArcGISAngularUnitId.Degrees), ArcGISGeodeticCurveType.Geodesic).Distance;
+                            UpdateDisplay();
+                        }
+                        featurePoints.Add(lastStop);
+                        // Interpolate middle points between last point and this point.
+                        Interpolate(lastStop, lineMarker, featurePoints);
+                        featurePoints.Add(lineMarker);
+                    }
+
+                    // Add this point to stops and also to feature points where stop is user-drawed, and feature points is a collection of user-drawed and interpolated.
+                    stops.Push(lineMarker);
+
+                    if (isPolygonMode)
+                    {
+                        if (featurePoints.Count > 3)
+                        {
+                            Interpolate(lineMarker, featurePoints[0], lastToStartInterpolationPoints);
+                        }
+                        CreateandCalculatePolygon();
+                    }
+                    if (featurePoints.Count >= 2)
+                        RenderLine(ref featurePoints);
```

Comment:
Formatting

Reply:
This doesn't relate to the touch controls added in this PR and can be roped into the General Samples Code Cleanup issue.

Reply:
This is copy and paste code from the Original Geometry sample which means this was either missed or not addressed when the Geometry Sample was under PR.